Libraries/VvVtkWidgets/CMakeLists.txt

```diff
@@ -27,6 +27,14 @@ set(vvVtkWidgetsSources
   vvVideoQueryDialogPrivate.cxx
 )
 
+if(CMAKE_CONFIGURATION_TYPES)
```

Comment:
Instead of checking `CMAKE_CONFIGURATION_TYPES`, please use the [`GENERATOR_IS_MULTI_CONFIG`](https://cmake.org/cmake/help/latest/prop_gbl/GENERATOR_IS_MULTI_CONFIG.html) global property.

Reply:
I saw that, but the documentation conveniently doesn't say what it's a property *on*, so I had no idea how to use it. Also, that would require bumping the minimum CMake version to at least 3.9?

Reply:
```cmake
get_property(GLOBAL _multi_config PROPERTY GENERATOR_IS_MULTI_CONFIG)
```

Reply:
> Also, that would require bumping the minimum CMake version to at least 3.9?

Indeed. If that is too much of a bump, then I suppose you could stick with `CMAKE_CONFIGURATION_TYPES`, knowing that if the user defines it on a single-config generator then things will break horribly.

Reply:
> I suppose you could stick with `CMAKE_CONFIGURATION_TYPES`, knowing that if the user defines it on a single-config generator then things will break horribly.

The user that does that is welcome to keep both pieces :wink:. Seeing as how there is no *point* setting that for non-MCG's, and this would hardly be the first place to use such a test...

Reply:
> this would hardly be the first place to use such a test...

Indeed, but we should discourage its use in new code. If the `cmake_minimum_required()` was closer to 3.9 already then I would be more adamant about this. Even 3.9 is old enough that I would tempted to make the bump if it was up to me.

Reply:
> I saw that, but the documentation conveniently doesn't say what it's a property _on_, so I had no idea how to use it. Also, that would require bumping the minimum CMake version to at least 3.9?

For what it's worth, if there is any VXL use (which it looks like there is) and you want C++11 support (which you do), then you will need CMake 3.9 or better anyway.

Reply:
Incidentally, this is Yet Another Reason why it would be nice if CMake had a real language; having to first fetch the property value rather than being able to name it directly is annoying.

Reply:
BTW, @KyleFromKitware, not sure if you caught my hint that the documentation could stand improvement? :slightly_smiling_face:

Reply:
I've opened [cmake/cmake#20413](https://gitlab.kitware.com/cmake/cmake/issues/20413) for discussion on how to improve this documentation.